### PR TITLE
feat: Added a python script utility to import logs stored in an S3 bucket to Seq 

### PR DESCRIPTION
**Pull request by AncientPatata**
# Motivation

This PR is related to the work done to [export logs from localhost to an S3 bucket](https://github.com/aneoconsulting/ArmoniK.Infra/pull/187). Adding a more general script to import saved ArmoniK logs from previous runs to a Seq deployment for further analysis. 

# Description

Building on the log2seq.py script from #62, generalized it to download entire directories that you can freely specify to match the new exports.

# Testing

Multiple test runs of the script.

# Impact

Not applicable.

# Additional Information

[Any additional information that reviewers should be aware of.]

# Checklist

- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] I have thoroughly tested my modifications and added tests when necessary.
- [ ] Tests pass locally and in the CI.
- [x] I have assessed the performance impact of my modifications.

### CODE REVIEW

File: tools/s3_to_seq/s3_to_seq.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+                ctx.logger.error(
+                    f"Attempt {attempt} at downloading file {obj.key} failed with error: \n {json.dumps(e.response['Error'])} "
+                )
+            ctx.logger.warning(f"All attempts at downloading {obj.key} failed, skipping..")
+    return file_list
+
+
+class LogSender:
+    def __init__(self, url: str):
+        """
```
Comment (qdelamea-aneo) [inline tools/s3_to_seq/s3_to_seq.py:162]: Move the docstring after the class definition.

File: tools/s3_to_seq/s3_to_seq.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    """
+    ctx = Context()
+    ctx.logger.debug("Sending gzipped json file : " + file_name)
+    with gzip.open(file_name, "r") as file:
+        with LogSender(url) as log_sender:
+            for line in file.read().decode("utf-8").split("\n"):
+                log_sender.sendlog(line)
+
+
+def main(args: argparse.Namespace):
```
Comment (qdelamea-aneo) [inline tools/s3_to_seq/s3_to_seq.py:252]: Do not use argument namespace as function input. Instead list arguments and then call the function using **args to unpack the namespace.

File: tools/s3_to_seq/s3_to_seq.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    logger = LoggingWrapper(args.log_level)
+    logger.info("S3 to Seq started")
+    context = Context(logger, Config(args.temp_path, args.nattempts))
+
+    logger.info("Temporary download directory set to " + args.temp_path)
+
+    try:
+        try:
+            s3 = boto3.client("s3")
+        except Exception:
```
Comment (qdelamea-aneo) [inline tools/s3_to_seq/s3_to_seq.py:262]: When handling exceptions you must always catch precise exceptions. Here you catch any exception even if it its not related to the client creation (like SIGTERM). So replace with the precise exeption(s) you wan to catch.

**Review by qdelamea-aneo — CHANGES_REQUESTED** [review]

**Review by aneojgurhem — COMMENTED** [review]
code is great but I would have expected to see this in ArmoniK instead of ArmoniK.Core

**Comment by AncientPatata** [comment]
> code is great but I would have expected to see this in ArmoniK instead of ArmoniK.Core

Moving it

**Pull request closed without merging**